service/history/commandChecker.go

```diff
@@ -173,6 +172,57 @@ func (c *workflowSizeChecker) checkIfMemoSizeExceedsLimit(
 	return nil
 }
 
+type childWorkflowLimitExceededErr string
+
+func (err childWorkflowLimitExceededErr) Error() string {
+	return strings.ToLower(string(err))
+}
+
+func (err childWorkflowLimitExceededErr) UserMessage() string {
+	return string(err)
+}
+
+func (c *workflowSizeChecker) checkIfNumChildWorkflowsExceedsLimit() error {
+	numPendingChildExecutionsStored := len(c.mutableState.GetPendingChildExecutionInfos())
+
+	errLimit := c.numPendingChildExecutionLimitError
+	if errLimit > 0 {
+		if numPendingChildExecutionsStored+1 > errLimit {
+			return childWorkflowLimitExceededErr(fmt.Sprintf(
+				"The number of pending child workflow executions is %d, "+
+					"and the limit established with \"%s\" is %d.",
+				numPendingChildExecutionsStored,
+				dynamicconfig.NumPendingChildExecutionLimitError,
+				errLimit,
+			))
+		}
+	}
+
+	warnLimit := c.numPendingChildExecutionLimitWarn
+	if warnLimit <= 0 {
+		return nil
+	}
+	if numPendingChildExecutionsStored+1 <= warnLimit {
+		return nil
+	}
+	if errLimit <= 0 {
+		c.logger.Warn(fmt.Sprintf(
+			"The number of pending child workflows, %d, "+
+				"is too high. Adding more child workflows could adversely affect performance.",
+			numPendingChildExecutionsStored,
+		))
+		return nil
+	}
+	c.logger.Warn(fmt.Sprintf(
```

Comment:
This log is for us. We should add a metrics here so we can alert on.

Reply:
This is for the user, not us. They created too many child workflows.

Reply:
what is the user? oss user?

Reply:
workflow user cannot see this log

Reply:
Fixed